lib/ruby_lsp/index.rb

```diff
@@ -0,0 +1,63 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  class Index
+    extend T::Sig
+    include Singleton
+
+    sig { returns(T::Array[String]) }
+    attr_reader :files
+
+    sig { returns(Requests::Support::PrefixTree) }
+    attr_reader :prefix_tree
+
+    sig { void }
+    def initialize
+      @files = T.let(collect_files, T::Array[String])
```

Comment:
This makes it a bit harder to type since we'll need to make the prefix tree nilable. Given that, outside of tests, we will never really want an empty index, are you okay with us moving forward like this?